spec/index.md

```diff
@@ -670,37 +680,122 @@ See [The system state tree](#state-tree) for details on the state tree.
 
-In order to make a query call to canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/query`. The request body consists of an authentication envelope with a `content` map with the following fields:
+In order to make a query call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/query`. The request body consists of an authentication envelope with a `content` map with the following fields:
 
--   `request_type` (`text`): Always `query`
+-   `request_type` (`text`): Always `"query"`
 
 -   `sender`, `nonce`, `ingress_expiry`: See [Authentication](#authentication)
 
 -   `canister_id` (`blob`): The principal of the canister to call.
 
--   `method_name` (`text`): Name of the canister method to call
+-   `method_name` (`text`): Name of the canister method to call.
 
--   `arg` (`blob`): Argument to pass to the canister method
+-   `arg` (`blob`): Argument to pass to the canister method.
 
-If the call resulted in a reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
+If the query call resulted in a reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
 
--   `status` (`text`): `replied`
+-   `status` (`text`): `"replied"`
 
 -   `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data.
 
 If the call resulted in a reject, the response is a CBOR map with the following fields:
 
--   `status` (`text`): `rejected`
+-   `status` (`text`): `"rejected"`
 
 -   `reject_code` (`nat`): The reject code (see [Reject codes](#reject-codes)).
 
 -   `reject_message` (`text`): a textual diagnostic message.
 
--   `error_code` (text): an optional implementation-specific textual error code (see [Error codes](#error-codes)).
+-   `error_code` (`text`): an optional implementation-specific textual error code (see [Error codes](#error-codes)).
+
+### Request: Signed query call {#http-signed-query}
+
+:::note
+
+The signed query call API is considered EXPERIMENTAL. Canister developers must be aware that the API may evolve in a non-backward-compatible way.
+
+:::
+
+A **signed** query call is a query call amended with signatures produced by IC nodes evaluating the query call to certify the query response.
+
+In order to make a signed query call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/signed_query`. The request body has the same structure as the request body for [Request: Query call](#http-query) with the `content` map having the following **additional** fields:
+
+-   (optional) `requested_signatures` (`text`): Determines whether and how many signatures are returned in the response; possible values are `"none"`, `"one"`.
+
+-   (optional) `requested_certificate` (`text`): Determines whether a certificate (as specified in [Certification](#certification)) is returned in the response; possible values are `"no"`, `"yes"`.
```

Comment:
"possible values are `"no"`, `"yes"`" why don't you make the field bool then ?

Reply:
I was thinking of adding more variants in the future, e.g., requesting specific paths to be present in the certificate.